Update mesh API documentation (#233)
nginxinc/nginx-service-mesh: docs/content/reference/api-usage.md

---
title: NGINX Service Mesh API

> description: "Instructions for using the NGINX Service Mesh API."

Reviewer: ```suggestion
description: "Instructions for interacting with the NGINX Service Mesh API."
```

Small change, but "using" is slightly more ambiguous: it could be referring to consuming information from the API or making changes with it.

toc: true
tags: ["api"]
weight: 200
categories: ["reference"]
docs: "DOCS-702"
---

## Overview

The NGINX Service Mesh API lives in a Kubernetes Custom Resource which allows a way to manage the global mesh configuration. This resource is created when the mesh is deployed, and can be updated at runtime via the Kubernetes API.

## Modifying the global mesh configuration

To update the global mesh configuration, use `kubectl` to edit the `meshconfig` resource that lives in the NGINX Service Mesh namespace. By default, the name of the resource is `nginx-mesh-config`.

```bash
kubectl edit meshconfig nginx-mesh-config -n nginx-mesh
```

> This will open your editor and allow for changes. To see the configurable fields, download the custom resource definition:

Reviewer: ```suggestion
This will open your default text editor and allow for changes. To see the configurable fields, download the custom resource definition:
```

Adding slightly more specificity. In context I presume this means `vi` or `nano` depending on the distribution.

{{< fa "download" >}} {{< link "crds/meshconfig.yaml" "`meshconfig-schema.yaml`" >}}

{{< warning >}}
If the `meshconfig` resource is deleted, or the `spec.meshConfigClassName` field is removed or changed, then the global mesh configuration cannot be updated, and unexpected behavior may occur.
{{< /warning >}}

## Viewing the global mesh configuration

> The `meshconfig` custom resource only contains configuration fields that can be changed at runtime. However, to view the full state of the mesh configuration, including fields that were set at installation, you can use the `nginx-meshctl` command line tool.

Reviewer: ```suggestion
The `meshconfig` custom resource only contains configuration fields that can be changed at runtime. To view the full state of the mesh configuration, including fields that were set at installation, you can use the `nginx-meshctl` command line tool.
```
